### PR TITLE
Add nmkr studio in builder tools

### PR DESCRIPTION
**Pull request by schaier-io**
## Checklist

- [x] I have read the [How to Contribute](https://developers.cardano.org/docs/portal-contribute/).
- [x] I have read the [Builder Tool Requirements](https://github.com/cardano-foundation/developer-portal/edit/staging/src/data/builder-tools.js)
- [x] I have run `yarn build` after adding my changes **without getting any errors**. 

## Builder Tool addition

* Title: *NMKR Studio*
* Description: *NMKR Studio removes the complexity of building Blockchain products. Making it easy to launch NFT collections* 
* Website:   [NMKR Studio](https://www.nmkr.io/studio)
* Getstarted: [Get started](https://studio.nmkr.io/register/)
* Tags:
  * `websocket`
  * `http`
  * `hosted`

### CODE REVIEW

**Comment by fill-the-fill** [comment]
I believe you're using some sort of code formatter that modified previously made changes including commas, spaces and etc. If you could revert that change and submit addition of builder tool only that would be amazing =)

File: src/data/builder-tools.js
Comment (rphair) [inline src/data/builder-tools.js:339]:
```suggestion
    getstarted: "https://docs.nmkr.io/NMKR-Studio/introduction",
```
`getstarted:` is intended to be a starting point for _learning_ about a tool, not for "getting started" with a _subscription_ to it 😜 This change seemed to be the best for new users in my opinion 🤔

**Review by rphair — COMMENTED** [review]
What @fill-the-fill says about not using a code editor that makes superfluous changes is absolutely essential.  If necessary you can use an ordinary text editor, like the ones that come with any OS, to simply add your ` title: "NMKR Studio"` section 🤓

Other than these issues I think your tool is a good candidate for the Showcase (contingent on your having a Free tier, which was I happy to see) and will approve it as soon as these preliminaries are taken care of 😎

File: src/data/builder-tools.js
Comment (schaier-io) [reply src/data/builder-tools.js]: Thanks for the quick response. I did the changes as suggested. I also noticed that NFT-MAKER under showcases is still under the old name, so I updated the naming and link. (former https://www.nft-maker.io/ now redirects to https://nmkr.io anyway, but to make it consistent and reflect the naming convention)

Comment (schaier-io) [reply src/data/builder-tools.js]: Sorry it seems as git showing it as different, that is however only because it is the original formatting (git compares it to the previous commit)

**Review by rphair — COMMENTED** [review]
With all the white space changes factored out, I can see from the metadata changes also on `showcases.js` (since nft-maker.io now redirects to nmkr.io) that this proposed new Builder Tool is already listed in Showcases (and a `favorite` at that): https://developers.cardano.org/showcase?tags=minting&tags=favorite

@rdlrt @os11k @katomm as I remember from previous discussion (https://github.com/cardano-foundation/developer-portal/pull/745 https://github.com/cardano-foundation/developer-portal/pull/800), duplicate listings were _verboten_.  Is there any reason that might not be applicable in this case?

- One reason to allow 2 separate listings (which I think could also have applied to Cexplorer) is that there are multiple aspects to this project: at least one of which is best for end users (with one Showcase URL) and also for developers / operators (with a different Builder Tool URL).

**Comment by schaier-io** [comment]
I would argue since we offer different products, to have them in different categories. Under dev tools we have NMKR Studio (which is our more developer/technical focused product, providing API but also low code integrations for websites of projects) The other part nmkr.io is more general, since we offer B2B solutions, but also various community tools to mint or for governance besides community tokens and other tools coming soon. If that does not qualify as unique listings, what do you think would be better fitting? Under Showcases or Devtools?

**Comment by rphair** [comment]
@schaier-io let's see what longer time editors @rdlrt @os11k @katomm think (I'm the newest of the lot).  There could be a historical reason to only allow one listing: I've only witnessed the recent part of that history so I can't say.  I would be willing to approve the 2 separate listings unless there is a consensus otherwise.  I'll ping them as well for review so we can see what they have to say 😎

**Comment by katomm** [comment]
I agree with the previous comments. It's already a good showcase and should stay that way in my opinion. Or it goes out of the showcase and goes into the builder tools, but for these tools, the tag "minting" was made in the showcase.

**Comment by schaier-io** [comment]
I removed the entry from build tools; So now only the renaming

File: src/data/builder-tools.js
Comment (schaier-io) [reply src/data/builder-tools.js:1023]: I just compared it to the current version in your branch https://github.com/cardano-foundation/developer-portal/blob/staging/src/data/builder-tools.js and it is exactly the same. Does it only show up as different because of the previous commits?

Comment (rphair) [reply src/data/builder-tools.js:1023]: It shows up at the same because whatever program you are using to compare them is ignoring the newline at the end of the file.

If in doubt about how to fix this, you can download a fresh copy of `src/data/builder-tools.js` in the `staging` branch and then do another commit and push.  Then you will see the message about having modified that file... by creating a slightly different white space syntax... disappear from this page that shows all the diffs: https://github.com/cardano-foundation/developer-portal/pull/865/files

This is a lot of work over a single character at the end of the file for both of us, but given the previous discussion I believe the commit history should _not_ show changes for a file that we've agreed above not to modify.

Comment (schaier-io) [reply src/data/builder-tools.js]: I am really sorry it took so long, the last time I copied the original document text and saved it. Git did not show any changes, therefore I could not commit and thought they were the same. Now replacing the file directly git noticed the difference. 

I hoped it is ok now

**Comment by fill-the-fill** [comment]
Last thing, could you please remove `yarn.lock` from your pr. If I'm not mistaken we've agreed to not commit `yarn.lock`. 

I could be wrong, @rphair could you please confirm?

After that I'm ready to approve as well :)

**Comment by os11k** [comment]
yarn.lock should not be committed, at least based on this:

https://developers.cardano.org/docs/portal-contribute#should-i-commit-the-yarnlock-file

**Comment by schaier-io** [comment]
I removed it, really sorry I included it in the last commit by accident

**Comment by rphair** [comment]
> @fill-the-fill: I could be wrong, @rphair could you please confirm?

After merging https://github.com/cardano-foundation/developer-portal/pull/832 I think we all observed that the automatic `yarn build` on GitHub wouldn't work if `yarn.lock` was deleted from the PR branch.  But if you read every comment in that discussion I think you'll agree that nobody was willing to step up and declare that officially.

That's probably why the section (https://developers.cardano.org/docs/portal-contribute/#should-i-commit-the-yarnlock-file) makes a statement that isn't true anymore... because it still refers to the time when `yarn` was only built locally.  If the PR were missing `yarn.lock` the GitHub script would have to backtrack and do a `yarn install`, and it didn't sound like the script were written that way.  I remember asking a question about that and thinking that only @oversize could answer it, but everyone including me was enthusiastic in first going ahead with the merge & seeing if it worked.

Anyway we can't test it here since `yarn.lock` __has not__ been removed from the branch at this time.  As I remember in comments with @oversize it's not a problem for a slightly different version of `yarn.lock` to be committed than the one that exists in `staging` or in other local PR branches.

The current standard we've all been using since #832... rightly or wrongly... has been for `yarn.lock` to exist in `staging` and therefore must also exist in every PR branch.  (cc @rdlrt @katomm in case I've missed something)

**Review by rphair — APPROVED** [review]
... in any case, approving now that superfluous white space has been removed from `builder-tools.js`.  Because of all the zig-zag changes I've also checked & verified that this builds locally. ✔️

FYI other reviewers especially @katomm may think, as I do, that the project image (designed for thumbnail resolution only) is too big.  But after all this is a `favorite` project so I don't want to give it too much of a hard time. 😵‍💫

**Comment by rphair** [comment]
... and drilling down in the "Details" of the GitHub CI "Yarn Build" tasks below, I can see it has an "Install" task that runs `yarn install` in the new container: so I am guessing @oversize would agree that it doesn't make a difference whether PR branches have a `yarn.lock` in them or not... since it'll be regenerated every time it's missing... right? 🤔

If that's true then maybe that section (https://developers.cardano.org/docs/portal-contribute/#should-i-commit-the-yarnlock-file) is true by default, since the only advantage to having `yarn.lock` in the PR branch is to save a few CPU seconds on a cloud software build that usually never be observed by anyone... only an announcement of whether it succeeds or fails.

So TL;DR I would argue that everything is therefore OK exactly the way it is. 🤠

**Comment by rdlrt** [comment]
The project itself isn't as big as to consume the github workflow time limits and we're not using github cache for NPM deployments. Hence , IMO - we should remove yarn.lock completely and add it to .gitignore.

**Review by rdlrt — APPROVED** [review]

**Review by os11k — APPROVED** [review]

**Comment by oversize** [comment]
The `yarn.lock` file should always be present (in the git repository). And therefore also be present in all branches. Yes everyone could always create a local version. But every local version could end up with some different dependencies. It is not about reducing the time needed to build the dependency tree but to allow everyone to end up with the same tree. 

The only time the yarn.lock file should change is when either a new dependency is added or the version of an existing dependency changes. As just happend in this pr: https://github.com/cardano-foundation/developer-portal/pull/862 

Yarn docs are actually pretty explicit about this: https://classic.yarnpkg.com/blog/2016/11/24/lockfiles-for-all/

Pipenv and composer recommend the same: 
https://getcomposer.org/doc/01-basic-usage.md#commit-your-composer-lock-file-to-version-control
https://github.com/pypa/pipenv/issues/598

Regarding our own docs: I would recommend to remove it completely. There should never be a reason to update the lock file in a pr that does not intentionally update depenencies. 

Should i open another issue where we can discuss this?

**Comment by rphair** [comment]
sure @oversize we should open an issue 🤓 since at this point it's a different subject than https://github.com/cardano-foundation/developer-portal/pull/832.  One question I would like to see answered, since I still don't know how to advise people after not getting a satisfactory answer to my question https://github.com/cardano-foundation/developer-portal/pull/832#issuecomment-1282654137, is (and it may be just my `git` deficiency that is the problem here):
- How do we advise contributors to remove `yarn.lock` from the files that will be committed with their PR once `yarn.lock` is again removed from [staging](https://github.com/cardano-foundation/developer-portal) (which currently _does_ have a `yarn.lock`) and other contributors' branches (e.g. [revamp-stake-pool-course](https://github.com/cardano-foundation/developer-portal/tree/revamp-stake-pool-course), which currently does not)?

**Comment by oversize** [comment]
Created a discussion for this: https://github.com/cardano-foundation/developer-portal/discussions/876

**Pull request merged**
